src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cache/FieldDataCacheRca.java

```diff
@@ -0,0 +1,204 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Field Data Cache RCA is to identify when the cache is unhealthy(thrashing) and otherwise, healthy.
+ * The dimension we are using for this analysis is cache eviction count, cache current weight(size) and
+ * cache max weight(size) configured.
+ * Note : For Field Data Cache, Hit and Miss metrics aren't available.
+ *
+ * <p>Cache eviction within Elasticsearch happens in following scenarios :
+ * <ol>
+ *   <li>Mutation to Cache (Entry Insertion/Promotion and Manual Invalidation)
+ *   <li>Explicit call to refresh()
+ * </ol>
+ *
+ * <p>The Cache Eviction requires that either the cache weight exceeds OR the entry TTL is expired.
+ * For Field Data Cache, no expire setting is present, so only in case of cache_weight exceeding the
+ * max_cache_weight, eviction(removal from Cache Map and LRU linked List, entry updated to EVICTED)
+ * happens.
+ *
+ * <p>Contrarily, the Cache Invalidation is performed manually on cache clear() and index close()
+ * invocation, with removalReason as INVALIDATED and a force eviction is performed to ensure cleanup.
+ *
+ * <p>This RCA reads 'fieldDataCacheEvictions', 'fieldDataCacheSize' and 'fieldDataCacheMaxSize'
+ * from upstream metrics and maintains a collector which keeps track of the time window period(tp)
+ * where we repeatedly see evictions for the last tp duration. This RCA is marked as unhealthy if
+ * tp is above the threshold(300 seconds) and cache size exceeds the max cache size configured.
+ *
+ */
+public class FieldDataCacheRca extends Rca<ResourceFlowUnit<HotNodeSummary>> {
+    private static final Logger LOG = LogManager.getLogger(FieldDataCacheRca.class);
+    private static final long EVICTION_THRESHOLD_TIME_PERIOD_IN_MILLISECOND = TimeUnit.SECONDS.toMillis(300);
+
+    private final Metric fieldDataCacheEvictions;
+    private final Metric fieldDataCacheSize;
+    private final Metric fieldDataCacheMaxSize;
+    private final int rcaPeriod;
+    private int counter;
+    private boolean exceedsSize;
+    protected Clock clock;
+    private final CacheEvictionCollector cacheEvictionCollector;
+
+    public <M extends Metric> FieldDataCacheRca(final int rcaPeriod, final M fieldDataCacheEvictions,
+                                                final M fieldDataCacheSize, final M fieldDataCacheMaxSize) {
```

Comment:
fieldDataCacheMaxSize will be collected as part node node config RCA. Do we need a separate metric here ?